core/modules/node/node.module

```diff
@@ -2912,6 +2927,40 @@ function _node_query_node_access_alter($query, $type) {
 }
 
 /**
+ * Returns the user's view_unpublished permissions in the following format:
+ *  $perms['any'] = TRUE if any of the below is TRUE, otherwise FALSE.
+ *  $perms['full'] = TRUE if "view any unpublished content" is set.
+ *  $perms['content_type'] = array of content types and the user's
+ *    associated view unpublished permission.
+*/
+function node_unpublished_user_permsissions() {
```

Comment:
Currently it looks like this was intended to have a much broader set of functionality that we're actually using. It's not actually called from anywhere yet, so I can't tell if it would be easy to use from a DX standpoint. If we need something like this, I would expect a function that directly checked the access by taking an $account and $node parameters and simply returning TRUE/FALSE.